apis/tempo/v1alpha1/tempostack_types.go

```diff
@@ -623,14 +623,23 @@ type QueryLimit struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Max Tags per User"
 	MaxBytesPerTagValues *int `json:"maxBytesPerTagValues,omitempty"`
-	// MaxSearchBytesPerTrace defines the maximum size of search data for a single
+
+	// DEPRECATED. MaxSearchBytesPerTrace defines the maximum size of search data for a single
 	// trace in bytes.
 	// default: `0` to disable.
 	//
 	// +optional
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Max Traces per User"
-	MaxSearchBytesPerTrace *int `json:"maxSearchBytesPerTrace"`
+	MaxSearchBytesPerTrace *int `json:"maxSearchBytesPerTrace,omitempty"`
+
+	// MaxSearchDuration defines the maximum search duration.
+	// If this value is not set, then spec.search.maxDuration is used.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Max Search Duration per User"
+	MaxSearchDuration metav1.Duration `json:"maxSearchDuration"`
```

Comment:
Why are we adding this field to the CR?

Reply:
This field is per-tenant, `.spec.search.maxDuration` is a global config option.
If we don't want it configurable, I can remove it.

It's the best "replacement" for the deprecated `MaxSearchBytesPerTrace` option which I could find ;)